### ISSUE TITLE
FieldProbe Point: Diagnostics Errors for Shape/Interpolation

### ISSUE THREAD

**Issue by jinze-liu** [ECP-WarpX/WarpX#4958]
When using input files for point diagnostics, I set ```algo.particle_shape = 3``` and ```interp_order = 1```. During execution, I encountered the following error:
```
1::Assertion `interp_order <= WarpX::nox' failed, file "/public/home/ljz_gpu/warpx/Source/Diagnostics/ReducedDiags/FieldProbe.cpp", line 155, Msg:
### ERROR   : Field probe interp_order should be less than or equal to
#            algo.particle_shape
!!!
SIGABRT
0::Assertion `interp_order <= WarpX::nox' failed, file "/public/home/ljz_gpu/warpx/Source/Diagnostics/ReducedDiags/FieldProbe.cpp", line 155, Msg:
### ERROR   : Field probe interp_order should be less than or equal to
#            algo.particle_shape
```
The backtrace file is as follows.
[Backtrace.0.txt](https://github.com/ECP-WarpX/WarpX/files/15492358/Backtrace.0.txt)

I tried modifying the values of the two variables, and the program only runs when ```interp_order = 0```, but the data from the point diagnostics are all zero.

Here is my input file：
[inputs.txt](https://github.com/ECP-WarpX/WarpX/files/15492380/inputs.txt)

Additionally, if I want to track the electromagnetic fields and current density at fixed grid points, as well as the velocities of fixed macroparticles, should I use Reduced Diagnostics?

**Comment by ax3l** [comment]
> Additionally, if I want to track the electromagnetic fields and current density at fixed grid points, as well as the velocities of fixed macroparticles, should I use Reduced Diagnostics?

You could use the *full* diagnostics and limit the space you want to output via `<diag_name>.diag_lo`, `<diag_name>.diag_hi` and other filter methods:
https://warpx.readthedocs.io/en/latest/usage/parameters.html#full-diagnostics

**Comment by ax3l** [comment]
Thank you for the reproducer! I see the same on my computer and somehow `WarpX::nox` is defined to `0` when the comparison happens. We'll dig deeper to fix this.

**Comment by ax3l** [comment]
Thank you for the report, fix coming in #4983.
